addons/sale/models/sale_order_line.py

```diff
@@ -1121,7 +1121,7 @@ def _prepare_invoice_line(self, **optional_values):
             res['analytic_distribution'] = self.analytic_distribution
         if analytic_account_id and not self.display_type:
             if 'analytic_distribution' in res:
-                res['analytic_distribution'][analytic_account_id] = res['analytic_distribution'].get(analytic_account_id, 0) + 100
+                res['analytic_distribution'][str(analytic_account_id)] = res['analytic_distribution'].get(str(analytic_account_id), 0) + 100
```

Comment:
What about the line below? Wouldn't it make sense to have it cast as a string too?
Maybe cast it on the declaration of `analytic_account_id` directly to avoid this issues.